add: Новый пистолет пулемет для СБ

vovakons opened the pull request:
## Что этот ПР делает

1. Добавляет третий вариант пушек в рандом оружейки.
2. Новый пистолет пулемет под 9 мм (*магазин на 24 патрона, урон 50% от урона 9мм патронов, может быть снаряжено леталом и резинками*)
3. Новая пушка имеет только режимы стрельбы одиночным или берстами, берсты на 3 патрона с минимальным временем между выстрелами в очереди. Точность новой пушки - винтовочная.

## Почему это хорошо для игры

Новая уникальная пушка для сб, не полностью автоматик, на распространенный калибр 9 мм.
Спрайты от кисика пушка.

## Демонстрация изменений

<details><summary>Демонстрации изменений</summary>
<p>

<img width="206" height="305" alt="image" src="https://github.com/user-attachments/assets/5ec2b7cc-1432-4282-af81-188b78f9790d" />

<img width="166" height="93" alt="image" src="https://github.com/user-attachments/assets/0f3525ee-b4e4-44e3-b8b0-85293164d666" />

</p>
</details>

## Тестирование

Локалка.

vovakons committed:
add new gun sparkle-a12

Добавлен новый пистолет пулемет Sparkle-A12 с магазином на 24 патронов 9 мм.
Добавлен рецепт нового магазина в автолат.

vovakons committed:
new random locker for armory

Новый вариант случайного ящика - с пистолет пулеметами Sparkle-A12.

vovakons committed:
sparkle-a12 modes fix

Добавлен модификатор урона 50% у новой пушки.
Теперь не фулл-авто, но стреляет берстами очень быстро.

vovakons committed:
linter fixes 1

Etrnlmelancholy commented:
Каждые 3 месяца по новой пушке для СБ, когда у них и так переполненный говной арсенал

vovakons committed:
linter fixes 2

vovakons committed:
integrated silencer and more damage for new gun

Добавлен неснимаемый глушитель.
Добавлена логика спавна оружия со стартовым модулем (просто добавить в
список).
Добавлены неснимаемые модули (настройка через флаг).
Увеличен урон с 50% до 70% у новой пушки.

SoulFN commented:
Лазерным 9мм тоже режет урон? А урон по стамине?

vovakons commented:
> Лазерным 9мм тоже режет урон? А урон по стамине?

Урон лазерам тоже режет до 70%.
Урон по стамине не режет пока.

vovakons commented:
> Каждые 3 месяца по новой пушке для СБ, когда у них и так переполненный говной арсенал

Да, это лучше чем бесполезная плюшевая игрушка с уникальным механом которым будут пользоваться полторы калеки.

vovakons committed:
new magazine for sec weapon

Новый спрайт магазина для пистолет пулемета.

vovakons committed:
sparkle a12 inhands

Спрайты инхендов.

littleboobs commented on code/game/objects/structures/crates_lockers/closets/secure/security.dm:
```diff
@@ -207,6 +207,28 @@
 	new /obj/item/gun/projectile/automatic/sp91rc(src)
 	new /obj/item/gun/projectile/automatic/sp91rc(src)
 
+/obj/structure/closet/secure_closet/guncabinet/sparkle_a12
+	name = "security Sparkle-A12 gun cabinet"
+	desc = "Защищённый шкаф для хранения пистолетов пулемётов Sparkle-A12. Шкаф прикручен к полу."
+	req_access = list(ACCESS_ARMORY)
+
+/obj/structure/closet/secure_closet/guncabinet/sparkle_a12/get_ru_names()
+	return list(
+		NOMINATIVE = "шкаф пистолет пулемётов Sparkle-A12",
+		GENITIVE = "шкафа пистолет пулемётов Sparkle-A12",
+		DATIVE = "шкафу пистолет пулемётов Sparkle-A12",
+		ACCUSATIVE = "шкаф пистолет пулемётов Sparkle-A12",
+		INSTRUMENTAL = "шкафом пистолет пулемётов Sparkle-A12",
+		PREPOSITIONAL = "шкафе пистолет пулемётов Sparkle-A12"
+	)
```
Название локализируй. Нет ничего лучше Орешников.

vovakons replied:
Может не надо? Итак нормальное название

littleboobs replied:
надо

littleboobs commented on code/game/objects/structures/crates_lockers/closets/secure/security.dm:
```diff
@@ -207,6 +207,28 @@
+	desc = "Защищённый шкаф для хранения пистолетов пулемётов Sparkle-A12. Шкаф прикручен к полу."
+	req_access = list(ACCESS_ARMORY)
+
+/obj/structure/closet/secure_closet/guncabinet/sparkle_a12/get_ru_names()
+	return list(
+		NOMINATIVE = "шкаф пистолет пулемётов Sparkle-A12",
+		GENITIVE = "шкафа пистолет пулемётов Sparkle-A12",
+		DATIVE = "шкафу пистолет пулемётов Sparkle-A12",
+		ACCUSATIVE = "шкаф пистолет пулемётов Sparkle-A12",
+		INSTRUMENTAL = "шкафом пистолет пулемётов Sparkle-A12",
+		PREPOSITIONAL = "шкафе пистолет пулемётов Sparkle-A12"
+	)
```
просто пустолет пулемёта

littleboobs commented on code/game/objects/structures/crates_lockers/closets/secure/security.dm:
```diff
@@ -207,6 +207,28 @@
 
+/obj/structure/closet/secure_closet/guncabinet/sparkle_a12
+	name = "security Sparkle-A12 gun cabinet"
+	desc = "Защищённый шкаф для хранения пистолетов пулемётов Sparkle-A12. Шкаф прикручен к полу."
```
ты разве не выносил прикручен к полу из описания?

vovakons replied:
я скопировал описание с ящиков вт и sp91

littleboobs commented on code/modules/projectiles/ammunition/magazines.dm:
```diff
@@ -533,6 +533,31 @@
 
+/obj/item/ammo_box/magazine/sparkle_a12
+	name = "Sparkle-A12 magazine (9mm)"
+	desc = "Магазин пистолет-пулемета \"Sparkle-A12\", заряженный патронами калибра 9 мм."
```
название

vovakons replied:
Считаю норм англ название оружия, не стоит прямо все на русский переводить

littleboobs replied:
ну здесь очевидная искра, это не сокращение какое-то

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -229,6 +229,33 @@
 	item_state = "SP-91-RC[magazine ? "-[get_ammo(FALSE) ? "20" : "0"]" : ""]"
 
 
+//Sparkle-A12
```
```suggestion
// MARK: Sparkle-A12
```

littleboobs commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -229,6 +229,33 @@
 
 
+//Sparkle-A12
+/obj/item/gun/projectile/automatic/sparkle_a12
+	name = "Sparkle-A12"
+	desc = "Компактный пистолет-пулемёт под калибр 9 мм."
+	icon_state = "sparkle-a12"
+	item_state = "sparkle-a12"
+	mag_type = /obj/item/ammo_box/magazine/sparkle_a12
+	fire_sound = 'sound/weapons/gunshots/1wt.ogg'
+	magin_sound = 'sound/weapons/gun_interactions/batrifle_magin.ogg'
+	magout_sound = 'sound/weapons/gun_interactions/batrifle_magout.ogg'
+	accuracy = GUN_ACCURACY_RIFLE
+	attachable_allowed = GUN_MODULE_CLASS_RIFLE_MUZZLE | GUN_MODULE_CLASS_RIFLE_RAIL | GUN_MODULE_CLASS_RIFLE_UNDER
+	attachable_offset = list(
+		ATTACHMENT_SLOT_MUZZLE = list("x" = 19, "y" = 3),
+		ATTACHMENT_SLOT_RAIL = list("x" = 4, "y" = 8),
+		ATTACHMENT_SLOT_UNDER = list("x" = 8, "y" = -5)
+	)
+	starting_attachment_types = list(/obj/item/gun_module/muzzle/suppressor/integrated)
+	recoil = GUN_RECOIL_MEDIUM
+	weapon_weight = WEAPON_HEAVY
+	fire_delay = 1
+	damage_mod = 0.7
```
рунеймсы потерял

vovakons committed:
localization fixes

PlayerUnknown14 commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -229,6 +229,43 @@
+// MARK: Sparkle-A12
+/obj/item/gun/projectile/automatic/sparkle_a12
+	name = "Sparkle-A12"
+	desc = "Компактный пистолет-пулемёт под калибр 9 мм."
```
```suggestion
	name = "A9 \"Sparkle\""
	desc = "Пистолет-пулемёт под калибр 9x19 мм, произведённый концерном \"Скарборо\". Штатно используется силовыми структурами \"Нанотрейзен\". Отличается надёжностью, высокой точностью и малыми габаритами. Предназначен для ближнего боя в условиях ограниченного пространства."
```
Слеши для переносов сам расставишь.

PlayerUnknown14 commented on code/modules/projectiles/guns/projectile/automatic.dm:
```diff
@@ -229,6 +229,43 @@
+		DATIVE = "пистолет пулемёту Sparkle-A12",
+		ACCUSATIVE = "пистолет пулемёт Sparkle-A12",
+		INSTRUMENTAL = "пистолет пулемётом Sparkle-A12",
+		PREPOSITIONAL = "пистолет пулемёте Sparkle-A12"
```
```suggestion
		NOMINATIVE = "А9 \"Искра\"",
		GENITIVE = "А9 \"Искра\"",
		DATIVE = "А9 \"Искра\"",
		ACCUSATIVE = "А9 \"Искра\"",
		INSTRUMENTAL = "А9 \"Искра\"",
		PREPOSITIONAL = "А9 \"Искра\""
```
9 типа 9 мм, пусть хоть что-то в этом названии будет иметь смысл. На чё буковку А заменить можно было бы — хз, так что пусть так будэ.

PlayerUnknown14 reviewed (changes requested):
Неймы не забудь у других объектов поменять потом.

vovakons committed:
localization

vovakons committed:
Merge branch 'master220' into new-sec-ballistic-weapon

vovakons committed:
new fucking linters fix
